pkg/schedule/checker/checker_controller.go

```diff
@@ -443,13 +488,46 @@ func (c *Controller) GetPauseController(name string) (*PauseController, error) {
 	}
 }
 
-func (c *Controller) updateTickerIfNeeded(ticker *time.Ticker) {
-	// Note: we reset the ticker here to support updating configuration dynamically.
-	newInterval := c.cluster.GetCheckerConfig().GetPatrolRegionInterval()
-	if c.interval != newInterval {
-		c.interval = newInterval
-		ticker.Reset(newInterval)
-		log.Info("checkers starts patrol regions with new interval", zap.Duration("interval", newInterval))
+// PatrolRegionContext is used to store the context of patrol regions.
+type PatrolRegionContext struct {
+	// workers
+	workersCtx    context.Context
+	workersCancel context.CancelFunc
+	regionChan    chan *core.RegionInfo
+	wg            sync.WaitGroup
+}
+
+func (p *PatrolRegionContext) init(ctx context.Context) {
+	p.regionChan = make(chan *core.RegionInfo, patrolRegionChanLen)
+	p.workersCtx, p.workersCancel = context.WithCancel(ctx)
+}
+
+func (p *PatrolRegionContext) stop() {
+	close(p.regionChan)
```

Comment:
It is better to add a failpoint to wait for all rest region consumed in `regionChan` before close regionChan, rather than `time.Sleep(100 * time.Millisecond)` in L171. This is more stable.

If possible, we can always wait it even if it is not in testing.

Reply:
`time.Sleep(100 * time.Millisecond)` in L171 is used to wait that regions to be consumed. 

```
co.PatrolRegions()
re.Empty(oc.GetOperators())
```

For example, if we enable this failpoint, it will wait 100 ms for goroutine consuming regions. And it will check `re.Empty(oc.GetOperators())` immediately after failpoint.

Reply:
I understand the role of L171. But I think sleep is a destabilizing factor. And it's fine to wait for it to finish consuming, or actively consume all regions before exiting and then exit.

Reply:
remove sleep, PTAL